lib/hobbes/boot/farray.hob

```diff
@@ -21,10 +23,12 @@ anyi f i xs =
     true
   else
     (anyi(f,i+1L,xs))
+{-# UNSAFE anyi #-}
 
 // [doc] determines whether any element of the array satisfied the predicate
 any :: (a -> bool, [a]) -> bool
 any f xs = anyi(f,0L,xs)
+{-# SAFE any #-}
```

Comment:
No way, every single function?  Whose idea was this and why is this needed?

Reply:
Yes, every single unsafe core function is annotated accordingly in this PR.
```sh
> anyi(\_.false,100l,[0..10])
fish: “hi” terminated by signal SIGSEGV (Address boundary error)
```
`any`, however, is safe

Reply:
I don't think that anyone is actually using `anyi`, but if they do then why not just fix the indexing bug?  What's the point of adding all of these annotations? What's the point of adding a `SAFE` annotation on `any`?  Who actually asked for this?